### ISSUE TITLE
Optimization: Remove 'accept-encoding: gzip' from default

### ISSUE THREAD

**Issue by joeconwaystk** [dart-lang/coverage#186]
Note: originally posted in another repo, not quite sure where to trace this back to, so adding it here since this is where the behavior manifests.

We are running into an issue where some heavy duty tests that spawn multiple isolates are failing to run coverage fast enough in Travis-CI (hits the 50 minute maximum).

Looking at CPU profiler, the coverage tool spends about 15% of runtime decompressing gzipped data (or at least in a method that is only triggered when decompressing data). Since the VM service is most often used intra-machine (?), it might be valuable to turn off gzip by default.

(It doesn't appear that the dart:io Websocket implementation allows this behavior yet.)

**Comment by joeconwaystk** [comment]
Here's an example log from Travis-CI: https://travis-ci.org/stablekernel/aqueduct/jobs/246279872.

**Comment by cbracken** [comment]
This looks doable via [CompressionOptions](https://api.dartlang.org/stable/2.1.0/dart-io/CompressionOptions-class.html) during websocket creation via [Websocket.fromUpgradedSocket](https://api.dartlang.org/stable/2.1.0/dart-io/WebSocket/WebSocket.fromUpgradedSocket.html).

I've filed vm_service_client#43, but will keep this open for the related package:coverage fix.

**Comment by cbracken** [comment]
Another alternative here is that we either bring back the hand-written VM Service client code that was removed in dart-lang/coverage@f3dfba4, or move to using the [vm_service_lib](https://github.com/dart-lang/vm_service_drivers) package, which would make it pretty simple for us to write our own connection code, then just pass that to the VMService constructor as happens in [vm_service_lib_io.dart](https://github.com/dart-lang/vm_service_drivers/blob/master/dart/lib/vm_service_lib_io.dart):

```dart
CompressionOptions options = ...;
WebSocket socket = await WebSocket.connect(wsUri, compression: options);
StreamController<String> controller = new StreamController();
socket.listen((data) => controller.add(data));
return new VmService(
    controller.stream, (String message) => socket.add(message),
    log: log, disposeHandler: () => socket.close());
```

**Comment by cbracken** [comment]
This was fixed in #258.
